spec/controllers/user_rule_effects_controller_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'spec_helper'
+
+RSpec.describe UserRuleEffectsController, type: :controller do
+  let(:workflow) { create :workflow }
+  let(:rule) { create :user_rule, workflow: workflow }
+
+  before { fake_session admin: true }
+
+  describe '#index' do
+    it 'lists effects for a rule' do
+      rules = [create(:user_rule_effect, user_rule: rule),
+                  create(:user_rule_effect, user_rule: rule)]
+
+      get :index, params: {workflow_id: workflow.id, user_rule_id: rule.id}, format: :json
+      json_response = JSON.parse(response.body)
+      expect(json_response.map { |i| i["id"] }).to match_array(rules.map(&:id))
+    end
+
+    it 'returns empty list when there are no user rules' do
+      get :index, params: {workflow_id: workflow.id, user_rule_id: rule.id}, format: :json
+      json_response = JSON.parse(response.body)
+      expect(json_response).to eq([])
+    end
+  end
+
+  describe "#show" do
+    it 'shows a particular effect' do
+      effect = create :user_rule_effect, user_rule: rule
+      get :show, params: { id: effect.id, user_rule_id: rule.id, workflow_id: workflow.id }, format: :json
+      result = JSON.parse(response.body)
+      expect(result["id"]).to eq(effect.id)
+    end
+
+    it 'returns 404 for an effect that doesnt exist' do
+      effect = create :user_rule_effect, user_rule: rule
+      get :show, params: {id: effect.id+100, user_rule_id: rule.id, workflow_id: workflow.id}, format: :json
+      expect(response.status).to eq(404)
+    end
+  end
+
+  describe "#create" do
+    it 'makes a new effect' do
+      post :create, params: {user_rule_effect: {action: 'promote_user', config: { workflow_id: 1234 }}, workflow_id: workflow.id, user_rule_id: rule.id }, format: :json
+
+      expect(response.status).to eq(200)
+      result = JSON.parse(response.body)
+      expect(result["id"]).not_to be(nil)
+      expect(result["user_rule_id"]).to eq(rule.id)
+    end
+
+    it 'redirects to the user rule in html mode' do
+      post :create, params: {user_rule_effect: {action: 'promote_user', config: { workflow_id: 1234 }}, workflow_id: workflow.id, user_rule_id: rule.id }, format: :html
+
+      expect(response.status).to eq(302)
+      location = response.headers["Location"]
+      expect(location).to end_with("workflows/#{workflow.id}/user_rules/#{rule.id}")
```

Comment:
These three lines could be shortened to `expect(response).to redirect_to(workflow_rule_path(workflow, rule))`